packages/@uppy/core/src/UIPlugin.ts

```diff
@@ -35,10 +39,20 @@ function debounce (fn) {
  *
  * For plugins without an user interface, see BasePlugin.
  */
-class UIPlugin extends BasePlugin {
-  #updateUI
+class UIPlugin<
+  Opts extends PluginOpts & { direction?: 'ltr' | 'rtl' },
+  M extends Meta,
+  B extends Body,
```

Comment:
Should we default to `any` to minimize the migration work?
```suggestion
  Opts extends PluginOpts & { direction?: 'ltr' | 'rtl' } = any,
  M extends Meta = any,
  B extends Body = any,
```

Reply:
It's on purpose so we see errors when things are untyped. Also one of my biggest issues during this refactor was that generics are not comparable and that was only fixed if you don't set defaults and always type them.